torch/onnx/symbolic_helper.py

```diff
@@ -419,6 +420,126 @@ def _interpolate_get_scales_and_mode(g, input, size, scale_factor, mode , align_
     return scale_factor, mode
 
 
+def _interpolate_common(name, dim, interpolate_mode):
+    def symbolic_fn(g, input, output_size, *args):
+        scales, align_corners = _get_interpolate_attributes(g, interpolate_mode, args)
+        align_corners = _maybe_get_scalar(align_corners)
+        coordinate_transformation_mode = "asymmetric" if interpolate_mode == "nearest" \
+            else "align_corners" if align_corners else "pytorch_half_pixel"
+
+        if scales is None:
+            input_size = g.op("Shape", input)
+            input_size_beg = _slice_helper(g, input_size, axes=[0], ends=[2], starts=[0])
+            output_size = g.op("Cast", output_size, to_i=cast_pytorch_to_onnx['Long'])
+            output_size = g.op("Concat", input_size_beg, output_size, axis_i=0)
+
+            if _has_more_optional_parameters():
+                empty_roi = _optional_input_placeholder_tensor(g)
+                empty_scales = _optional_input_placeholder_tensor(g)
+            else:
+                empty_roi = g.op("Constant", value_t=torch.tensor([], dtype=torch.float32))
+                empty_scales = g.op("Constant", value_t=torch.tensor([], dtype=torch.float32))
+
+            return g.op("Resize",
+                        input,
+                        empty_roi,
+                        empty_scales,
+                        output_size,
+                        coordinate_transformation_mode_s=coordinate_transformation_mode,
+                        cubic_coeff_a_f=-0.75,  # only valid when mode="cubic"
+                        mode_s=interpolate_mode,  # nearest, linear, or cubic
+                        nearest_mode_s="floor")  # only valid when mode="nearest"
+        else:
+            if _has_more_optional_parameters():
+                empty_roi = _optional_input_placeholder_tensor(g)
+            else:
+                empty_roi = g.op("Constant", value_t=torch.tensor([], dtype=torch.float32))
+
+            return g.op("Resize",
+                        input,
+                        empty_roi,
+                        scales,
+                        coordinate_transformation_mode_s=coordinate_transformation_mode,
+                        cubic_coeff_a_f=-0.75,  # only valid when mode="cubic"
+                        mode_s=interpolate_mode,  # nearest, linear, or cubic
+                        nearest_mode_s="floor")  # only valid when mode="nearest"
+    return symbolic_fn
+
+
+def __interpolate_common(g, input, size, scale_factor, mode, align_corners, recompute_scale_factor):
```

Comment:
Please rename to __interpolate_helper to keep the existing style.

Reply:
Sure